### PR TITLE
Rating Page

### PR DESCRIPTION
**Pull request by butalas**
### Description
This branch designs the rating page.

### Context
It mimics the designs from Figma as one of the screens for the app. I wasn't sure whether to do the tool bar so I didn't implement that on the screen.

### Issue
#27 

### Screenshot
img_6058

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,11 +1,44 @@
 import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import { StyleSheet, Text, View, Dimensions, Image} from 'react-native';
```
Comment (ijemmao) [inline App.js]: Revert your App.js file back to what it looks like on the `master` branch

File: screens/login.js [hunk 2]
```diff
@@ -1,16 +1,18 @@
 import React from 'react';
```
Comment (ijemmao) [inline screens/login.js]: It looks like you checked out this branch from your `login` branch - please make sure that you checkout only from the `master` branch. If you don't, you include code that isn't technically approved to be merged into another.

Remove at files that are connected to login - that includes other js or assets files.

File: screens/rating.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+const window = Dimensions.get("window")
+
+export default class Login extends React.Component {
+  state = {
+    fontLoaded: false,
+  };
+  async componentDidMount() {
+    await Font.loadAsync({
+      'Montserrat-Regular': require('/Users/nona/shopu/assets/fonts/Montserrat-Regular.otf'),
```
Comment (ijemmao) [inline screens/rating.js]: Again, change the paths of your fonts

File: screens/rating.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+            <View style={styles.topBanner} />
+            <Image style={styles.shoppingCarticon}
+                source={require('/Users/nona/shopu/images/shopping-cart.png')} />
+            <Image style={styles.personIcon}
+                source={require('/Users/nona/shopu/images/profile_person.png')} />
+            <Text style={styles.thankYouforText}>Thank U for</Text>
+            <Text style={styles.usingText}>using</Text>
+            <Text style={styles.shopuText}>shopU!</Text>
+            <View style={styles.ratingBox} />
+            <Image style={styles.ratingIcons}
```
Comment (ijemmao) [inline screens/rating.js]: This can't just be images - this needs to be individual stars

Instead of having just an `Image` component, you should use the `View` component, and set its background to the star image.

File: screens/rating.js (hunk 3 above)
Comment (butalas) [reply screens/rating.js]: When I change the paths and reload the app, it says it's unable to find the fonts. Do I have to put in exactly `./../assets/fonts/Montserrat-Regular.otf`?

File: screens/rating.js (hunk 4 above)
Comment (butalas) [reply screens/rating.js]: So how do I split the stars from that image? I saved the rating_group.png as a component from Figma. Do I need to make 20 Views?

File: screens/rating.js (hunk 3 above)
Comment (ijemmao) [reply screens/rating.js]: No, that was just an example

When I was walking through bash commands, I mentioned that `.` is the current directory and `..` is the parent directory.

Comment (butalas) [reply screens/rating.js]: I got it to work! Thanks!

**Comment by butalas** [comment]
I addressed your comments on the rating page. I used margins for the stars, but the remaining components on the screen are still positioned absolutely.

**Comment by ijemmao** [comment]
@butalas This is looking really good - the only thing I want you to do is remove the `Font.loadAsync()` code from the rating page.

**Before you do that**: on your local `ratingPage` branch call `git pull origin ratingPage --force` this will sync up your local with the remote version

**Comment by butalas** [comment]
Sorry I haven't been on campus all day so I just looked at the font thing. I removed the `Font.loadAsync()` code from the rating page like you asked.

File: screens/buyer_search.js [hunk 5]
```diff
@@ -0,0 +1,206 @@
+import React from 'react';
```
Comment (ijemmao) [inline screens/buyer_search.js]: This file shouldn't be included in this PR, you can remove it

File: screens/login.js [hunk 6]
```diff
@@ -1,38 +1,24 @@
...
```
Comment (ijemmao) [inline screens/login.js]: Remove this file

File: screens/rating.js [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+            <Image style={styles.shoppingCarticon}
+                source={require('./images/shopping-cart.png')} />
+            <Image style={styles.personIcon}
+                source={require('./images/profile_person.png')} />
+            <Text style={styles.thankYouforText}>Thank U for</Text>
+            <Text style={styles.usingText}>using</Text>
+            <Text style={styles.shopuText}>shopU!</Text>
+            <View style={styles.ratingBox} />
+            <Text style={styles.delivererText}>Deliverer:</Text>
+            <View style={{width: 300, flexDirection: 'row', marginTop: 800, marginLeft: 185}}>
```
Comment (ijemmao) [inline screens/rating.js]: You shouldn't have inline styles - make a new style below

Also use flexboxes - you shouldn't be using `800px` margins to position items

File: screens/rating.js [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+              {this.stars()}
+              </View>
+            <Text style={styles.timingText}>Timing:</Text>
+            <View style={{width: 300, flexDirection: 'row', marginTop: 65, marginLeft: 185}}>
```
Comment (ijemmao) [inline screens/rating.js]: This line is repeated three times - create a style

File: screens/rating.js [hunk 9]
```diff
@@ -0,0 +1,224 @@
...
+      )
+    })
+    return starMap;
+  }
+
+  render() {
+    return (
+      <View>
+      {
+        this.state.fontLoaded ? (
```
Comment (ijemmao) [inline screens/rating.js]: Remote this line - `App.js` already checks for the font

File: screens/rating.js [hunk 10]
```diff
@@ -0,0 +1,224 @@
...
+import { Font } from 'expo';
+import FontLoad from './components/FontLoad';
+
+const window = Dimensions.get("window")
+
+export default class Rating extends React.Component {
+  state = {
+    fontLoaded: false,
+  };
+  componentWillMount = () => {
```
Comment (ijemmao) [inline screens/rating.js]: Remove the `componentWillMount` function

File: screens/rating.js [hunk 11]
```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import { StyleSheet, Text, View, Dimensions, Image} from 'react-native';
+import { Font } from 'expo';
+import FontLoad from './components/FontLoad';
+
+const window = Dimensions.get("window")
+
+export default class Rating extends React.Component {
+  state = {
```
Comment (ijemmao) [inline screens/rating.js]: Remove this `state` variable

File: screens/rating.js [hunk 12]
```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import { StyleSheet, Text, View, Dimensions, Image} from 'react-native';
+import { Font } from 'expo';
+import FontLoad from './components/FontLoad';
```
Comment (ijemmao) [inline screens/rating.js]: Remove this line

File: screens/rating.js [hunk 13]
```diff
@@ -0,0 +1,224 @@
+import React from 'react';
+import { StyleSheet, Text, View, Dimensions, Image} from 'react-native';
+import { Font } from 'expo';
```
Comment (ijemmao) [inline screens/rating.js]: Remove this line - `App.js` should be the only file with references to `Font`

File: screens/rating.js (hunk 7 above)
Comment (butalas) [reply screens/rating.js]: Should I make it come down like 3/4 of the page (something along those lines)?

Comment (butalas) [reply screens/rating.js]: Like a percentage instead of pixels

Comment (ijemmao) [reply screens/rating.js]: I would first use flexboxes first and then use margins to tweak the small details

**Comment by butalas** [comment]
I just addressed all the comments you made regarding the rating page!
